Automattic/jetpack
# Allow single-quotes when bypassing Deferred JS

Fixes #22268

#### Changes proposed in this Pull Request:

* Update regular expression to include single quotes, double quotes, and no quotes attribute while ignoring deferred js.
* After this change users can exclude javascript deferring with the following formats:
```
<script data-jetpack-boost="ignore"></script>
<script data-jetpack-boost='ignore'></script>
<script data-jetpack-boost=ignore></script>
```

#### Jetpack product discussion


NA

#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






* Add a basic plugin with the following content to a Jetpack Boost enabled website:
```
function add_test_script() {
	echo '<script>console.log( "No ignore attribute" );</script>';
	echo '<script data-jetpack-boost="ignore">console.log( "Ignore attribute in double quotes" );</script>';
	echo "<script data-jetpack-boost='ignore'>console.log( 'Ignore attribute in single quotes' );</script>";
	echo '<script data-jetpack-boost=ignore>console.log( "Ignore attribute in no quotes" );</script>';
}
add_action( 'wp_head', 'add_test_script' );
```
* Enable the `Defer Non-Essential JavaScript` option in Jetpack Boost dashborad
* Open the website homepage
* Make sure that all the scripts with `ignore` attribute do not differ, and only the script without the `ignore` attribute differs.
